src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,37 +32,70 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        try{
+        	splitAddressStringIntoComponents(trimmedAddress);
         }
-        this.value = trimmedAddress;
+        catch (IndexOutOfBoundsException e){
+        	throw new IllegalValueException(MESSAGE_ADDRESS_ILLEGAL_FORMAT);
+        }
+        if (!isValidAddress()) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_ILLEGAL_FORMAT);
+        }
+    }
+    
+    /**
+     * Splits address into component classes e.g. Block, Street, Unit PostalCode
+     * 
+     */
+    private void splitAddressStringIntoComponents(String value) throws IndexOutOfBoundsException{
+    		String[] strArr = value.split(",");
+    	this.block = new Block(strArr[INDEX_BLOCK].trim());
+    	this.street = new Street(strArr[INDEX_STREET].trim());
+    	this.unit = new Unit(strArr[INDEX_UNIT].trim());
+    	this.postalCode = new PostalCode(Integer.valueOf(strArr[INDEX_POSTALCODE].trim()));
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public boolean isValidAddress() {
+        return Block.isValidBlockNumber(this.block.getNumber()) && PostalCode.isValidPostalCode(this.postalCode.getPostalCode());
+    	//return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
     @Override
     public String toString() {
-        return value;
+    	//return value;
+        return this.block.getNumber() + ", " + this.street.getStreet() + ", " + this.unit.getUnit() + ", " + Integer.toString(this.postalCode.getPostalCode());
```

Comment:
Could apply this functionality you mentioned to the next tutorial of Inheritance